### DISCUSSION TITLE
Mail::alwaysTo not working for synced mails

### DISCUSSION THREAD

**Discussion by vatsake** [laravel/framework#53727] [Q&A] [answered]
### Laravel Version

11.24.1

### PHP Version

8.3.6

### Description

In my AppServiceProvider boot method I have
```
if (!App::isProduction()) {
    Mail::alwaysTo('John.Doe@mail.com');
}
```

I have a job that sends an email.
```
class SendEmail implements ShouldQueue
{
    use Dispatchable;
    use InteractsWithQueue;
    use Queueable;
    use SerializesModels;

    /**
     * Create a new job instance.
     */
    public function __construct(public array|string $from, public string $subject = '', public string|array $to = [], public array|string $cc = [], public array|string $bcc = [], public string $content = '')
    {
        //
    }

    /**
     * Execute the job.
     */
    public function handle(): void
    {
        Mail::html($this->content, function (Message $message) {
            $message->to($this->to)
                ->bcc($this->bcc)
                ->cc($this->cc)
                ->subject($this->subject);
            if (is_array($this->from)) {
                $message->from(...$this->from);
            } else {
                $message->from($this->from);
            }
        });
    }
}
```

### Steps To Reproduce

If sending email like this:
`dispatch(new SendEmail(subject: 'test', to: 'Not.John.Doe@mail.com', content: 'test'));`
The email will be sent to alwaysTo address.

If dispatching like this:
`dispatch_sync(new SendEmail(subject: 'test', to: 'Not.John.Doe@mail.com', content: 'test'));`
The email will **NOT** be sent to alwaysTo address...

BTW I'm using octane, so it might be 1 reason why it doesn't work.

**Comment by crynobone** [comment]
> BTW I'm using octane, so it might be 1 reason why it doesn't work.

Yes, this is due to Octane as the service providers will only load once while the mailers get reset on each request.

**Comment by munezaclovis** [comment, marked as answer]
@vatsake I think if you might be able to fix this if you directly set this on the config, this would always work as expected since Laravel loads the mail config anytime a mailer gets created

```
if(!app()->isProduction){
    config()->set('mail.to', ['address' => 'John.Doe@mail.com', 'name' => 'John Doe'])
}
```

I haven't tested this so you might wanna check

**Reply by vatsake** [reply]
Thanks, that did the trick.
But I think this should be fixed in Laravel or Octane.
